EE-2274 - Added Assays, Experiments and Analyses objects

M-casado opened the pull request:
Done:
- [ ] Added a Run object, that was later fused with ArrayAssay into Assays. 
- [ ] Added Experiment object, fusing both sequencing experiment and ArrayExperiment.
- [ ] Added Analysis object.
- [ ] Added loci descriptors.
- [ ] Updated protocols. 
- [ ] Added new folder with controlled vocabulary (CV) lists (that are commonly too extensive to have in the common definitions)
- [ ] Updated nodes in which both the term and CURIE were requested when the term was part of a CV list. Requesting the term was enough if the ``meta:enum`` array had the documented CURIEs. 

To do (to ease the review):
- [ ] Add changes to the CHANGELOG.
- [ ] Update JSON markdowns.

M-casado commented:
As special mentions for the review:

1. Fields that could be redundant? For instance, I checked things like experimental design and assay types, all the ontologies I could get a hold of, and still thought those would be useful together. 
2. Additional types of analysis and experiment? I did some research regarding the experiment and analysis types, but still there may be some that I missed and could be added. 
3. Requirements? There may be fields that one could say should be required and are not, and the other way around.

afoix reviewed (approved):
It looks good to me @M-casado 🙂

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+              "title": "Type of analysis",
+              "description": "Overall type of the analysis. Term chosen from a controlled vocabulary (CV) list. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+              "enum": [ "sequence variation", "sequence alignment", "phenotype characterization", "processed sequencing data", "processed array data", "sequence annotation", "sequence assembly" ],
+              "meta:enum": [ "sequence variation__ERO:0100211 - Analysis of variations at specific loci in the genomes of organisms (mutation and polymorphism) across or within a species, population, or individual (e.g healthy vs diseased tissue).", "sequence alignment__ERO:0100032 - objective to display graphically how the sequences of two or more macromolecules align along a linear axis.", "phe ...
```
```suggestion
              "meta:enum": [ "genetic sequence variation analysis__ERO:0100211 - Analysis of variations at specific loci in the genomes of organisms (mutation and polymorphism) across or within a species, population, or individual (e.g healthy vs diseased tissue).", "sequence alignment__ERO:0100032 - objective to display graphically how the sequences of two or more macromolecules align along a linear axis.", "phenotype characterization__ERO:0000923 - The result of an organismal assay that involves characterization of a phenotype; any observable characteristic or trait of an organism: such as its morphology, development, biochemical or physiological properties, behavior, and products of behavior (such as a bird's nest). Phenotypes result from the expression of an organism's genes as well as the influence of environmental factors and the interactions between the two.", "processed sequencing data__ - Analysis in which raw sequencing data (e.g. FastQ files) were processed in any way (e.g. normalization, noise reduction, alignment...) and transformed into processed genotype data files [EFO:0004663].", "processed array data__ - Analysis in which raw array data (e.g. CEL files) were processed in any way (e.g. normalization, noise reduction...) and transformed into processed array data files [EFO:0004096].", "sequence annotation__ - For submitting sequence annotation files, which are usually 'tab' files. Examples include gene count and OTU tables from metagenomic studies.", "sequence assembly__topic:0196 - The assembly of fragments of a DNA sequence to reconstruct the original sequence." ]            }
```

malloryfreeberg replied:
Suggest that if references to ontology curies are to be used, that the associated text is also used. E.g. if citing ERO:0100211 then ensure the associated text, "genetic sequence variation analysis" is used.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+              "type": "string",
+              "title": "Type of analysis",
+              "description": "Overall type of the analysis. Term chosen from a controlled vocabulary (CV) list. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+              "enum": [ "sequence variation", "sequence alignment", "phenotype characterization", "processed sequencing data", "processed array data", "sequence annotation", "sequence assembly" ],
```
This enum list is fine for now. In the future, we can switch from a enum list of individual values to providing a higher-level ontology term and allowing any value under that term. For example, under "[sequence analysis objective](http://purl.obolibrary.org/obo/OBI_0200081)".

malloryfreeberg replied:
Actually, the terms "processed sequencing data" and "processed array data" don't seem like they belong here. These terms are referencing a type of data, not a type of analysis. 

Maybe add "gene expression"?

M-casado replied:
Added gene expression and moved the types of data elsewhere 👍

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+        },
+        "allOf": [
+          {
+            "title": "If the analysis type includes a sequence comparison, the reference is expected",
+            "if": {
+              "required": ["analysis_types"],
+              "properties": {
+                "analysis_types": {
+                  "type": "array",
+                  "minItems": 1,
+                  "contains": {
+                    "type": "string",
+                    "enum": [ "sequence variation", "sequence alignment"]
```
I'm not sure what type of analysis this block is referencing. If it means any type of analysis where an alignment step is done, one could also be doing a sequence annotation or a sequence assembly analysis.

M-casado replied:
But since it's a constraint placed on those that **always** have a reference, I only enumerated Seq. variation and alignment. Correct me if I'm wrong, but can't assemblies and annotations be _de novo_ (without a reference)?

malloryfreeberg replied:
- Assemblies --> can be done by aligning experiments reads to a reference OR aligning experiment reads to themselves
- Annotations --> can be done with or without a reference
- Variation --> can be done with a reference (what variation is in my samples compared to a known reference) OR without a reference (what variation exists among all my sequenced samples)
- Alignment --> can be done with a reference (e.g. HISAT) OR without a reference (e.g. clustering/aligning reads to each other, not a reference, to quantify )

So really, none of these analysis approaches _requires_ a reference. They can all be done without one. It may be easy in this first round to remove this constraint.

M-casado replied:
Well the constraint only applies for ``sequence variation`` and ``sequence alignment`` but it's true that there are cases in which the external reference of an assembly is not needed. 

But even if the assembly reference is not always used, the sequence against which some other sequences are compared is always there. Should we modify the ``reference_alignment_details`` node to have some kind of explanation (e.g. what sequence is used as ref. if not an assembly), or expect that in the protocols for an alignment data transformation?

malloryfreeberg replied:
I suggest making this filed more conservative / less restrictive. Provide an option for the user to supply a reference, if they had one, but don't but constraints on the assumptions because science is messy :)

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+        "type": "array",
+        "title": "Loci of the targeted genomic feature",
+        "description": "Array of items that unambiguously define the loci of targeted genomic features in the analysis. For example, if the aim of the analysis was to detect variants in the human gene TAF1 and TP53, their identifiers will be expected in two items of this array.",
+        "minItems": 1,
```
```suggestion
```

malloryfreeberg replied:
This attribute must not be required. Some analyses, e.g. whole genome sequencing to detect genomic variation, so not have targeted genomic features.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+
+      "experimental_designs": {
+        "type": "array",
+        "title": "Experimental designs of the analysis [EFO:0001426]",
```
It's confusing why some titles have ontology curies included and some do not. What is trying to be achieved here?

M-casado replied:
I'm trying to add context not only to the ``enum`` values of the metadata, but to the documentation itself. When they are present it means I found one close enough to the term I wanted to represent in the metadata. 

As discussed, these will be included in a different keyword: ``"meta:..."`` at the object's level.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+        "description": "Type of raw data the analysis uses to obtain the processed files.",
+        "additionalProperties": false,
+        "uniqueItems": true,
+        "minItems": 1,
```
Is this attribute required under the current schema? If not, I suggest not making it required here.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+        "description": "Custom attributes of an analysis: reusable attributes to encode tag-value pairs (e.g. Tag being 'internal tag' and its Value 'this analysis is corresponds to internal tag XYZ') with optional units. Its properties are inherited from the common-definitions.json schema.",
+        "additionalProperties": false,
+        "uniqueItems": true,
+        "minItems": 1,
```
This attribute should not be required.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+        "type": "array",
+        "title": "Loci of the targeted genomic feature",
+        "description": "Array of items that unambiguously define the loci of targeted genomic features in the experiment. For example, if the experiment aim was to detect variants in the human gene TAF1 and TP53, their identifiers will be expected in two items of this array.",
+        "minItems": 1,
```
As with analysis, this attribute should not be required.

Actually, I think this attribute should be removed from analysis. The loci being targeted are presumably targeted during the experiment, and not the analysis. I'm not sure how the values would be different for this attribute between the analysis and experiment object, so it makes sense to keep it only in 1 object.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+        "type": "string",
+        "title": "Assayed molecule of the experiment [EFO:0002772]",
+        "description": "The human readable ID (e.g. DNA assay), chosen from a list of controlled vocabulary (CV), of the assayed molecule. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+        "enum": ["DNA assay", "RNA assay", "Metabolomic profiling", "Protein assay"],
```
These enum values are types of assays rather than types of molecules. I suggest using the list of molecules listed under the [EFO:0004446](http://www.ebi.ac.uk/efo/EFO_0004446) ontology.

M-casado replied:
Although I agree that being more specific could be beneficial, under EFO:0004446 ontology there are no references to synthetic DNA, proteins, metabolites...

Would it be better to just add DNA and RNA, and leave other categories out? Another thing we could do is something similar to the *_type and *_subtype classifications we used in other places: having a general molecule identifier (DNA, RNA, metabolite and protein) and a subtype (e.g. messenger RNA).

malloryfreeberg replied:
I suggest having just a type (also having subtype is too complex right now) listing all the molecules under [EFO:0004446](http://www.ebi.ac.uk/efo/EFO_0004446) (maybe ignore the "adhesion factor") and also add two terms for protein and metabolite (although we really shouldn't get this data)

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+        "description": "Custom attributes of an experiment: reusable attributes to encode tag-value pairs (e.g. Tag being 'additional description' and its Value 'this experiment is a re-do of another 3 experiments due to...') with optional units. Its properties are inherited from the common-definitions.json schema.",
+        "additionalProperties": false,
+        "uniqueItems": true,
+        "minItems": 1,
```
This attributes should not be required.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+      "sample_number": {
+        "type": "integer",
+        "title": "Number of samples of the analysis",
+        "description": "Number of samples included in the analysis (i.e. analysed together). One sample corresponds to one biological replicate [EFO:0002091] (e.g. genetic content from a single cell, a tissue, buccal swab, etc.) from a single individual or from several individuals. Shall not be mistaken for technical replicates [EFO:0002090] being used several times (see https://www.ebi.ac.uk/seq ...
```
I would remove the link to Confluence.

M-casado replied:
Moved diagram to GH and changed URL in ``sample_grouping``.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+        "examples": [ "The analysis was conducted with the objective of... ...and for that purpose we compared untreated controls against..." ]
+      },
+
+      "sample_number": {
+        "type": "integer",
+        "title": "Number of samples of the analysis",
+        "description": "Number of samples included in the analysis (i.e. analysed together). One sample corresponds to one biological replicate [EFO:0002091] (e.g. genetic content from a single cell, a tissue, buccal swab, etc.) from a single individual or from several individuals. Shall not be mistaken for technical replicates [EFO:0002090] being used several times (see https://www.ebi.ac.uk/seq ...
+        "examples": [ 30 ]
+      },
```
Actually, I would just remove this for now, and put a feature in the backlog to investigate better ways to represent sample number.

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
     "meta:version": "0.0.0",
     "description": "Empty for the moment.",
-    "required": [""],
+    "required": ["object_id", "sample_number", "analysis_protocols", "experimental_designs", "analysis_type_specifications" ],
```
```suggestion
    "required": ["object_id", "experimental_designs", "analysis_type_specifications" ],
```

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+
+      "experimental_designs": {
+        "type": "array",
+        "title": "Experimental designs of the analysis [EFO:0001426]",
```
```suggestion
        "title": "Experimental designs of the analysis",
        "meta:attribute_curie": "EFO:0001426"
```

M-casado replied:
Done 👍

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+              "type": "string",
+              "title": "Type of analysis",
+              "description": "Overall type of the analysis. Term chosen from a controlled vocabulary (CV) list. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+              "enum": [ "sequence variation", "sequence alignment", "phenotype characterization", "processed sequencing data", "processed array data", "sequence annotation", "sequence assembly" ],
```
```suggestion
              "enum": [ "sequence variation", "sequence alignment", "phenotype characterization", "sequence annotation", "sequence assembly", "gene expression" ],
```

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+      "types_of_data": {
+        "type": "array",
+        "title": "Type of data [IAO:0000100]",
+        "description": "Type of raw data the analysis uses to obtain the processed files.",
```
```suggestion
        "description": "Type of input data the analysis uses to obtain the processed files.",
```

malloryfreeberg commented on schemas/EGA.analysis.json:
```diff
@@ -5,9 +5,190 @@
+        }
+      },
+
+      "experimental_designs": {
```
Possibly redundant with a similar term at the level of the study or dataset. Suggest not having "design"-type attributes at the analysis object level.

M-casado replied:
These will be moved to the to the study/dataset 👍

malloryfreeberg commented on schemas/controlled_vocabulary_schemas/EGA.cv.assay_type_by_sequencer.json:
Only because I know EGA already has data from these assay types, I would suggest minimally adding the following:
- single cell sequencing (EFO_0007832)
- RNA-seq (EFO_0008896)
- exome sequencing (EFO_0005396)
- single nucleus RNA sequencing (EFO_0009809)
- genotyping by high throughput sequencing (EFO_0002771)
- methylation profiling by high throughput sequencing (EFO_0002761)

In the future, we should allow any term under "[assay by sequencer](http://www.ebi.ac.uk/efo/EFO_0003740)"

M-casado replied:
All of them are already listed (with their ontologies) as Assay Subtypes (e.g. ``RNA-seq`` or ``genotyping by high throughput sequencing`` subtypes are of type ``assay by high throughput sequencer``).

And I think there were [assays by molecule](http://www.ebi.ac.uk/efo/EFO_0002772) that were not listed within assay by sequencer, so we should also keep that in mind when accepting only the hierarchy of  [assay by sequencer](http://www.ebi.ac.uk/efo/EFO_0003740).

malloryfreeberg commented on schemas/controlled_vocabulary_schemas/EGA.cv.assay_subtype_by_sequencer.json:
```diff
@@ -0,0 +1,10 @@
+    "title": "EGA Controlled Vocabulary (CV) for assay subtypes by sequencer [EFO:0003740]",
+    "meta:version": "0.0.0",
+    "description": "Controlled Vocabulary (CV) list for assay subtypes by sequencer: any ontologized term for a subtype (i.e. child ontology) of an sequencing assay [EFO:0003740].",
+    "enum": [ "bulk immune repertoire sequencing", "HTGTS-Rep-seq", "Ig-Seq", "Paired VH:VL Antibody Repertoire Analysis", "Rep-Seq", "single cell immune repertoire sequencing", "10x immune profiling", "10x Ig enrichment", "10x TCR enrichment", "TCR Chain Paring", "MC-Hi-C", "CUT&RUN", "DNA-seq", "ATAC-seq", "Fast-ATAC", "scATAC-seq", "10x multiome", "10x scATAC-seq", "scATAC-seq (Microfluidics)" ...
```
Having the assay type and assay subtype attributes with specific CVs is OK for the first draft of the schemas. In the future, we should consider ways to take advantage of existing ontologies, which have already classified and structured all these terms. We should also consider ways to make it easy for users to provide a meaningful value. For example, users can provide the most specific term that describes their assay type, and we can infer more general classifications using ontology tools.

M-casado replied:
Yes, for now I used CVs because we have not implemented automated ontology validations. As discussed we could also add already tips for the future like ``"meta:child_node_of": ...``.

Creating these CVs is tiresome, it's even worse when users need to suggest additions all the time, and we should just use ontologies hierarchies, definitions and descriptions. We could already dedicate time to: (1) reproduce what Biovalidator did (in-situ ontology validation); (2) find a way to download the current hierarchy and place that as a CV (saves time for the moment and shouldn't take that much time as the first option). 

Regarding the last part (inferring general classifications), it does take some time and effort on our side to infer them, and what I suggest is to also request for that general classification from the users. That way we won't be wondering if we chose correctly or not what they meant.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
-    "description": "Empty for the moment.",
-    "required": [""],
+    "description": "Metadata schema used by the European Genome-phenome Archive (EGA) to validate its experiment metadata object. An experiment is considered a planned and intentionally designed process performed as part of a study. Further details can be found in the EGA-metadata-schema GitHub repository (https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas) and EGA-archive website (h ...
+    "required": ["object_id", "sample_number", "experiment_protocols", "assay_technology", "experimental_designs", "assay_type_descriptor", "assayed_molecule", "experiment_type_specifications"],
```
```suggestion
    "required": ["object_id", "experiment_protocols", "assay_technology", "experimental_designs", "assay_type_descriptor", "assayed_molecule", "experiment_type_specifications"],
```

malloryfreeberg replied:
Wondering if it's confusing that some of the attributes in the experiment object are named `assay_*`, although they are not in the assay object. Perhaps all attributes in experiment object should start with `experiment_*`.

M-casado replied:
I don't have strong opinions on neither options, since I agree that having experiment_* is more straightforward, but having assay_technology aligns better with the attribute's definition and meaning in the ontology.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+        "examples": [ "The experiment was conducted with the objective of... ...and for that purpose we compared untreated controls against..." ]
+      },
+
+      "sample_number": {
+        "type": "integer",
+        "title": "Number of samples of the experiment",
+        "description": "Number of samples included in the Assay (i.e. pooled into one single microarray, labelled differently). One sample corresponds to one biological replicate [EFO:0002091] (e.g. genetic content from a single cell, a tissue, buccal swab, etc.) from a single individual or from several individuals. Shall not be mistaken for technical replicates [EFO:0002090] being used several t ...
+        "examples": [ 30 ]
+      },
+
```
I suggest removing this attribute. We should instead calculate sample number from the submitted metadata. By also having this attribute, we risk having inconsistency within the metadata.

M-casado replied:
Done: https://www.ebi.ac.uk/panda/jira/browse/EE-2344

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+
+      "assay_technology": {
+        "title": "Technology used in the assay",
+        "description": "Technology used in the assay. This node allows for an easy filtering of the technology (e.g. a sequencer Illumina NextSeq 500) used to obtain the raw data (e.g. sequence files) in an assay.",
```
```suggestion
        "description": "Technology used in the assay. This node allows for an easy filtering of the technology used to obtain the raw data (e.g. sequence files) in an assay.",
        "examples" : ["sequencer Illumina NextSeq 500"],
```
```suggestion
        "description": "Technology used in the assay. This node allows for an easy filtering of the technology (e.g. a sequencer Illumina NextSeq 500) used to obtain the raw data (e.g. sequence files) in an assay.",
```
```suggestion
        "description": "Technology used in the assay. This node allows for an easy filtering of the technology (e.g. a sequencer Illumina NextSeq 500) used to obtain the raw data (e.g. sequence files) in an assay.",
```

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+
+      "assay_technology": {
+        "title": "Technology used in the assay",
+        "description": "Technology used in the assay. This node allows for an easy filtering of the technology (e.g. a sequencer Illumina NextSeq 500) used to obtain the raw data (e.g. sequence files) in an assay.",
```
```suggestion
        "description": "Technology used in the assay. This node allows for an easy filtering of the technology used to obtain the raw data (e.g. sequence files) in an assay.",
        "examples" : ["sequencer Illumina NextSeq 500"],
```

malloryfreeberg replied:
Minor suggestion: have examples explicitly as a separate field for the attribute. This is more clear for users, gives us more flexibility for providing this information in different forms and for different uses in the future (e.g. we can auto-generate example JSONs by filling in with terms from "examples".

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+      "assayed_molecule": {
+        "type": "string",
+        "title": "Assayed molecule of the experiment [EFO:0002772]",
+        "description": "The human readable ID (e.g. DNA assay), chosen from a list of controlled vocabulary (CV), of the assayed molecule. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
```
```suggestion
        "description": "The biomolecule type assayed in the experiment. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
        "examples": ["genomic DNA", "messenger RNA"],
```

malloryfreeberg replied:
"human readable" is redundant - everything should be human-readable :P

M-casado replied:
True, I used that phrase some times when trying to refer to the human's term for a concept or identifier, something that they can understand. It is indeed redundant in this case 👍

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+      "types_of_data": {
+        "type": "array",
+        "title": "Types of data [IAO:0000100]",
+        "description": "Types of data the experiment encompasses.",
```
```suggestion
        "description": "Types of data the experiment generates.",
```

malloryfreeberg replied:
I think for experiment, you want the type of data the experiment produces? As opposed to the type of data an analysis uses as input? 

I'm thinking maybe this attribute is redundant with other attributes. For example, if you indicate `"assayed_molecule": "genomic DNA"` and relevant values for `assay_technology` and `assay_type_descriptor`, what value does indicating `"types of data": ["genomic data"]` add?

M-casado replied:
Well, the molecule would not be ``genomic DNA`` but ``DNA`` alone. That way ``genomic data`` does add some information on whether the DNA is from a metagenomic conglomerate, synthetic DNA, or from a genome.

M-casado replied:
Depending on what changes we apply to the assayed_molecule object my previous comment will make sense or not.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+      "experiment_type_specifications": {
+        "type": "object",
+        "title": "Experiment type specifications",
+        "description": "Node containing different sets of fields that depend on the specific experiment type. The main categories of experiment types follow a similar pattern as the assay technology: either array experiments (those in which an [array instrument [EFO:0002698]](http://www.ebi.ac.uk/efo/EFO_0002698) was used) or sequencing experiments (those in which a [sequencing instrument [EFO:00 ...
```
```suggestion
        "description": "Node containing additional attributes to describe the experiment, either array experiments (those in which an [array instrument [EFO:0002698]](http://www.ebi.ac.uk/efo/EFO_0002698) was used) or sequencing experiments (those in which a [sequencing instrument [EFO:0003739]](http://www.ebi.ac.uk/efo/EFO_0003739) was used). For example, if an array was used, its Array Design Format (ADF) will be expected.",
```

malloryfreeberg replied:
Some descriptions, like this one, are a little verbose, so I suggested slimming down the description to keep it specifically describing the attribute.

malloryfreeberg commented on schemas/EGA.assay.json:
```diff
@@ -0,0 +1,327 @@
+    "title": "EGA assay metadata schema",
+    "meta:version": "0.0.0",
+    "description": "Metadata schema used by the European Genome-phenome Archive (EGA) to validate its assay metadata object. This object is intended to contain metadata about the raw qualitative or quantitative test performed to determine any kind of biological property of a biological sample. It can be of different types: (1) sequencing assay [EFO:0003740] (e.g. sequence CRAM or FastQ files); an ...
+    "required": ["object_id", "sample_number", "assay_type_specifications", "assay_files"],
```
```suggestion
    "required": ["object_id", "assay_type_specifications", "assay_files"],
```

M-casado replied:
done

malloryfreeberg commented on schemas/EGA.assay.json:
```diff
@@ -0,0 +1,327 @@
+        "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/EGA-ISO-date-YYYY-MM-DD-pattern"
+      },
+
+      "sample_number": {
```
Suggest removing sample_number, as discussed in previous comments

M-casado replied:
done

M-casado replied:
But just for the record, we shall always take into account the sample_grouping to count samples linked to other objects.

malloryfreeberg commented on schemas/EGA.assay.json:
```diff
@@ -0,0 +1,327 @@
+
+      "assay_files": {
+        "type": "array",
+        "title": "Sequencing files from an assay",
```
```suggestion
        "title": "Data files produced from an assay",
```

malloryfreeberg replied:
Generalised to "Data files" given the files might come from an array assay, as well.

M-casado replied:
Done

malloryfreeberg reviewed (changes requested):
Review done. Lots of great work here!

There are some requests for changes, some suggestions that can be further discussed, and some recommendations/thoughts for future changes. Happy to do another review after changes are made.

M-casado commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,331 @@
+        ]
+      },
+
+      "assayed_molecule_descriptor": {
+        "type": "object",
+        "title": "Descriptor of the assayed molecule",
+        "description": "Node containing information about the assayed molecule: the material entity (e.g. DNA) that was used to generate the data.",
+        "additionalProperties": false,
+        "required": ["assayed_molecule_type"],
+        "properties": {
+          "assayed_molecule_type": {
+            "type": "string",
+            "title": "Type of the assayed molecule of the experiment",
+            "description": "The assayed molecule (e.g. DNA), chosen from a list of controlled vocabulary (CV). If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+            "enum": ["DNA", "RNA", "metabolie", "protein"],
+            "meta:enum": ["DNA__CHEBI:16991", "RNA__CHEBI:33697", "metabolite__EFO:0004727", "protein__CHEBI:36080"]
+          },
+          "assayed_molecule_subtype": {
+            "type": "string",
+            "title": "Subtype of the assayed molecule of the experiment",
+            "description": "Name of the assayed molecule subtype. We highly recommend the usage of names given to ontologized molecule subtypes, specially those at the [Experimental Factor Ontology (EFO)](https://www.ebi.ac.uk/ols/ontologies/efo).",
+            "examples": ["cDNA", "genomic DNA", "mitochondrial DNA", "messenger RNA", "ncRNA", "insulin", "serum albumin" ]
+          },
+          "assayed_molecule_subtype_curie": {
+            "type": "string",
+            "title": "CURIE of the assayed molecule subtype",
+            "description": "Ontology term in CURIE format (e.g. 'PR:000003918') of the assayed molecule subtype. Search for the ontologized term at the [Ontology Lookup Service (OLS)](https://www.ebi.ac.uk/ols/index).",
+            "examples": ["PR:000003918", "EFO:0008481", "EFO:0008479", "EFO:0008480", "CHEBI:33699"],
+            "allOf": [
+              {
+                "title": "General pattern of a CURIE",
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/curie_general_pattern"
+              }
+            ]
+          }
+        }
+      },
+
```
This is what I meant, @malloryfreeberg, with [my comment](https://github.com/EbiEga/ega-metadata-schema/pull/34#discussion_r820587615)

M-casado commented:
@malloryfreeberg I modified the JSON schemas a bit based on both new additions, fixes and your feedback:

- [ ] Removed additional labels from dataset's and study's objects (that allowed for free-form terms)
- [ ] Modified the sampling details, having both the date and the collection site
- [ ] Modified the experimental designs (a.k.a. study designs): removed from experiment, analyses and datasets; now exclusively at study level.
- [ ] Also separated dataset and study types (previously merged), closer to the ENA's way. 
- [ ] Modified the types of data: now there are input and output type fields in analyses, and output one in experiment. This includes the processed data files (sequencing and array), which were removed from analyses types.
- [ ] Properties' CURIEs within titles were moved to its new attribute ``meta:property_curie``. 
- [ ] Fixed examples from previous ticket (datasets, policies and datasets).
- [ ] Modified the assayed molecule field, having new subtypes and types. 
- [ ] Removed confluence link and added that image within the GH repo. 
- [ ] Fixed validation examples based on changes.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+        "type": "string",
+        "title": "Assayed molecule of the experiment [EFO:0002772]",
+        "description": "The human readable ID (e.g. DNA assay), chosen from a list of controlled vocabulary (CV), of the assayed molecule. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+        "enum": ["DNA assay", "RNA assay", "Metabolomic profiling", "Protein assay"],
```
```suggestion
        "enum": ["DNA", "cDNA", "genomic DNA", "mitochondrial DNA", "RNA", "messenger RNA", "ncRNA", "non polyA RNA", "long non polyA RNA", "nuclear RNA", "polyA RNA", "long polyA RNA", "snRNA", "total RNA", "adhesion factor", "metabolite", "protein"],
```

malloryfreeberg replied:
Since we don't get many of these types of assays (protein data should go to PRIDE and metabolite data to MetaboLights), in the first instance it is OK to just have the generic terms:

"metabolite__EFO:0004727"
"protein__CHEBI:36080"

M-casado replied:
Hmmmm my last commit changed that node, are you taking a look at previous versions of the files?

M-casado commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,334 @@
+        ]
+      },
+
+      "assayed_molecule_descriptor": {
+        "type": "object",
+        "title": "Descriptor of the assayed molecule",
+        "description": "Node containing information about the assayed molecule: the material entity (e.g. DNA) that was used to generate the data.",
+        "additionalProperties": false,
+        "required": ["assayed_molecule_type"],
+        "properties": {
+          "assayed_molecule_type": {
+            "type": "string",
+            "title": "Type of the assayed molecule of the experiment",
+            "description": "The assayed molecule (e.g. DNA), chosen from a list of controlled vocabulary (CV). If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+            "enum": ["DNA", "RNA", "metabolie", "protein"],
+            "meta:enum": ["DNA__CHEBI:16991", "RNA__CHEBI:33697", "metabolite__EFO:0004727", "protein__CHEBI:36080"]
+          },
+          "assayed_molecule_subtype": {
+            "type": "string",
+            "title": "Subtype of the assayed molecule of the experiment",
+            "description": "Name of the assayed molecule subtype. We highly recommend the usage of names given to ontologized molecule subtypes, specially those at the [Experimental Factor Ontology (EFO)](https://www.ebi.ac.uk/ols/ontologies/efo).",
+            "examples": ["cDNA", "genomic DNA", "mitochondrial DNA", "messenger RNA", "ncRNA", "insulin", "serum albumin" ]
+          },
+          "assayed_molecule_subtype_curie": {
+            "type": "string",
+            "title": "CURIE of the assayed molecule subtype",
+            "description": "Ontology term in CURIE format (e.g. 'PR:000003918') of the assayed molecule subtype. Search for the ontologized term at the [Ontology Lookup Service (OLS)](https://www.ebi.ac.uk/ols/index).",
+            "examples": ["PR:000003918", "EFO:0008481", "EFO:0008479", "EFO:0008480", "CHEBI:33699"],
+            "allOf": [
+              {
+                "title": "General pattern of a CURIE",
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/curie_general_pattern"
+              }
+            ]
+          }
+        }
+      },
```
@malloryfreeberg This is the current version of that node, having the type and then an open subtype in which one can tell whether the RNA is a messenger RNA, LncRNA, etc.

malloryfreeberg replied:
I think this is too complex.

malloryfreeberg commented on schemas/EGA.experiment.json:
```diff
@@ -2,12 +2,324 @@
+        "title": "Assayed molecule of the experiment [EFO:0002772]",
+        "description": "The human readable ID (e.g. DNA assay), chosen from a list of controlled vocabulary (CV), of the assayed molecule. If you cannot find your term in the CV list, please create an issue at our [metadata GitHub repository](https://github.com/EbiEga/ega-metadata-schema) proposing its addition.",
+        "enum": ["DNA assay", "RNA assay", "Metabolomic profiling", "Protein assay"],
+        "meta:enum": ["DNA assay__EFO:0001456", "RNA assay__EFO:0001457", "Metabolomic profiling__EFO:0000752", "Protein assay__EFO:0001458"]
```
```suggestion
        "meta:enum": ["DNA__CHEBI:16991", "cDNA__EFO:0008481", "genomic DNA__EFO:0008479", "mitochondrial DNA__EFO:0008480", "RNA__CHEBI:33697", "messenger RNA__CHEBI:33699", "ncRNA__SO:0000655", "non polyA RNA__EFO:0005017", "long non polyA RNA__EFO:0005018", "nuclear RNA__EFO:0030052", "polyA RNA__OBI:0000869", "long polyA RNA__EFO:0005019", "snRNA__SO:0000274", "total RNA__EFO:0004964", "metabolite__EFO:0004727", "protein__CHEBI:36080"]
```

malloryfreeberg commented:
Review done :) Looks great. Just 3 outstanding comments:

- [ ] Suggestion to only have 1 type attribute for the biomolecule being assayed with the following terms (and I think you're already making this change):
```
terms: "DNA", "cDNA", "genomic DNA", "mitochondrial DNA", "RNA", "messenger RNA", "ncRNA", "non polyA RNA", "long non polyA RNA", "nuclear RNA", "polyA RNA", "long polyA RNA", "snRNA", "total RNA", "metabolite", "protein"

curies: ["DNA__CHEBI:16991", "cDNA__EFO:0008481", "genomic DNA__EFO:0008479", "mitochondrial DNA__EFO:0008480", "RNA__CHEBI:33697", "messenger RNA__CHEBI:33699", "ncRNA__SO:0000655", "non polyA RNA__EFO:0005017", "long non polyA RNA__EFO:0005018", "nuclear RNA__EFO:0030052", "polyA RNA__OBI:0000869", "long polyA RNA__EFO:0005019", "snRNA__SO:0000274", "total RNA__EFO:0004964", "metabolite__EFO:0004727", "protein__CHEBI:36080"]
```
- [ ] Suggestion for removing the requirement to have a reference for the two indicated analysis types (see [comment](https://github.com/EbiEga/ega-metadata-schema/pull/34/files#r824679878))
- [ ] Request to modify the valid study example to show how one could use this to indicate the study is related to COVID-19. This would enable us to test a real use case (tagging of COVID-19 studies with MONDO:0100096) that we know we want to support in the near future.

M-casado commented:
Regarding your last review, @malloryfreeberg 

> Suggestion to only have 1 type attribute for the biomolecule being assayed

I planned to do this in the beginning, but the issue is that the ontology is probably not complete, and thus there are RNA and DNA subtypes that we might lose. To avoid this one would also add ``DNA`` or ``RNA`` to the list, the "types", but this would lead, in my opinion, to the following inconsistency: if we mix both subtypes (e.g. ``cDNA``) and types (e.g. ``DNA``), since one encompasses the other, people may not know which one to provide. For example, if my DNA was of a type not listed in the ontology (e.g. ``viral DNA``?), they would pick the parent term (``DNA``), but if they have a ``cDNA`` they would pick a subtype. And in some cases submitters may pick the parent types even though they have the subtypes (e.g. "my cDNA is DNA, afteral, so I'll say ``DNA``").

> Suggestion for removing the requirement to have a reference for the two indicated analysis types

I agree, but see my [response](https://github.com/EbiEga/ega-metadata-schema/pull/34/files#r824766343), in case we add something else.

> Request to modify the valid study example to show how one could use this to indicate the study is related to COVID-19

We don't have a tag system yet in the schemas, although I had it before (the ``additional_keywords_list`` you mentioned. Would you have the COVID-19 tag there, or have it elsewhere like in the description, title, etc. as a submitter may provide it?

M-casado commented:
@malloryfreeberg Addressed the changes in my last three commits. 

- [ ] Added more study designs and the ``study_type`` is now an array: ``study_types``. Added ``COVID-19`` there and I expect it to be like a study-type keyword list. We can add more terms there, limit the amount of keywords given, etc. 
- [ ] Lifted restriction of reference alignment on analyses. 
- [ ] Changed the assayed molecule as you suggested.

Although regarding this last one I am a bit confused: 

>  In the specific case of viral DNA, if this was the sample then 1) it shouldn't be in EGA and 2) if it is, it's either the virus' genomic DNA or cDNA derived from a viral RNA genome.

I can think of cases in which a viral DNA could also be considered sensitive and personal data, and thus be welcomed at EGA; and I'm not entirely sure that viral DNA falls under the current description of [genomic DNA](http://www.ebi.ac.uk/efo/EFO_0008479) (``...normal chromosomal complement of...``), even if there are viral genomes. 

In any case, let me know if you want to check the changes before I change the CHANGELOG and create the markdowns.

malloryfreeberg commented:
> I can think of cases in which a viral DNA could also be considered sensitive and personal data, and thus be welcomed at EGA; and I'm not entirely sure that viral DNA falls under the current description of [genomic DNA](http://www.ebi.ac.uk/efo/EFO_0008479) (...normal chromosomal complement of...), even if there are viral genomes.

I guess we can just wait until we have a use case for this and use that to drive our decision of what to do here (which shouldn't be hard to change now that we manage the schemas).

Everything else LGTM! Great job and a monumental effort! 👍

malloryfreeberg reviewed (approved):
LGTM!
